components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/util/FilterUtil.java

```diff
@@ -0,0 +1,200 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.resource.collection.mgt.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.api.resource.collection.mgt.constant.APIResourceCollectionManagementConstants;
+import org.wso2.carbon.identity.api.resource.collection.mgt.exception.APIResourceCollectionMgtClientException;
+import org.wso2.carbon.identity.api.resource.collection.mgt.exception.APIResourceCollectionMgtServerException;
+import org.wso2.carbon.identity.api.resource.collection.mgt.model.APIResourceCollection;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.model.ExpressionNode;
+import org.wso2.carbon.identity.core.model.FilterTreeBuilder;
+import org.wso2.carbon.identity.core.model.Node;
+import org.wso2.carbon.identity.core.model.OperationNode;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.api.resource.collection.mgt.util.APIResourceCollectionManagementUtil.handleClientException;
+import static org.wso2.carbon.identity.api.resource.collection.mgt.util.APIResourceCollectionManagementUtil.handleServerException;
+
+/**
+ * Utility class for filter operations.
+ */
+public class FilterUtil {
+
+    /**
+     * Get the filter node as a list.
+     *
+     * @param filter Filter string.
+     * @throws APIResourceCollectionMgtClientException Error when validate filters.
+     */
+    private static List<ExpressionNode> getExpressionNodes(String filter)
+            throws APIResourceCollectionMgtClientException {
+
+        List<ExpressionNode> expressionNodes = new ArrayList<>();
+        filter = StringUtils.isBlank(filter) ? StringUtils.EMPTY : filter;
+        try {
+            if (StringUtils.isNotBlank(filter)) {
+                FilterTreeBuilder filterTreeBuilder = new FilterTreeBuilder(filter);
+                Node rootNode = filterTreeBuilder.buildTree();
+                setExpressionNodeList(rootNode, expressionNodes);
+            }
+            return expressionNodes;
+        } catch (IOException | IdentityException e) {
+            throw handleClientException(
+                    APIResourceCollectionManagementConstants.ErrorMessages.ERROR_CODE_INVALID_FILTER_FORMAT);
+        }
+    }
+
+    /**
+     * Set the node values as a list of expression nodes.
+     *
+     * @param node       Filter node.
+     * @param expression List of expression nodes.
+     */
+    private static void setExpressionNodeList(Node node, List<ExpressionNode> expression)
+            throws APIResourceCollectionMgtClientException {
+
+        if (node instanceof ExpressionNode) {
+            if (StringUtils.isNotBlank(((ExpressionNode) node).getAttributeValue())) {
```

Comment:
We can combine two if statements

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5807.